core/src/main/java/io/substrait/relation/RelChildVisitor.java

```diff
@@ -0,0 +1,282 @@
+            return expr;
+          }
+
+          @Override
+          public Expression.I64Literal visit(Expression.I64Literal expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.FP32Literal visit(Expression.FP32Literal expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.FP64Literal visit(Expression.FP64Literal expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.StrLiteral visit(Expression.StrLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.BinaryLiteral visit(Expression.BinaryLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.TimeLiteral visit(Expression.TimeLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.DateLiteral visit(Expression.DateLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.TimestampLiteral visit(Expression.TimestampLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.TimestampTZLiteral visit(Expression.TimestampTZLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.IntervalYearLiteral visit(Expression.IntervalYearLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.IntervalDayLiteral visit(Expression.IntervalDayLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.UUIDLiteral visit(Expression.UUIDLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.FixedCharLiteral visit(Expression.FixedCharLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.VarCharLiteral visit(Expression.VarCharLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.FixedBinaryLiteral visit(Expression.FixedBinaryLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.DecimalLiteral visit(Expression.DecimalLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.MapLiteral visit(Expression.MapLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.ListLiteral visit(Expression.ListLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.StructLiteral visit(Expression.StructLiteral expr) {
+            return expr;
+          }
+
+          @Override
+          public Expression.Switch visit(Expression.Switch expr) {
+            expr.defaultClause().accept(this);
```

Comment:
If this is a rewrite, shouldn't we compose a new switch statement (and others below such as ifthen and scalarfunctioninvocation) so that any mutation remains the same? Otherwise, it's weird to be returning something from the visitor.

Reply:
Yes, but I was planning on doing that in my next commit. Didn't want to merge it with this one which is two commits: one for a bug fix (plus new test) and one for a deep traversal visitor (not yet rewrite).